mars/oscar/backends/pool.py

```diff
@@ -1029,19 +1027,26 @@ async def monitor_sub_pools(self):
         try:
             while not self._stopped.is_set():
                 for address in self.sub_processes:
-                    process = self.sub_processes[address]
-                    recover_events_discovered = (address in self._recover_events)
-                    if not await self.is_sub_pool_alive(process):  # pragma: no cover
-                        if self._on_process_down is not None:
-                            self._on_process_down(self, address)
-                        self.process_sub_pool_lost(address)
-                        if self._auto_recover:
-                            await self.recover_sub_pool(address)
-                            if self._on_process_recover is not None:
-                                self._on_process_recover(self, address)
-                    if recover_events_discovered:
-                        event = self._recover_events.pop(address)
-                        event.set()
+                    try:
+                        process = self.sub_processes[address]
+                        recover_events_discovered = (address in self._recover_events)
+                        if not await self.is_sub_pool_alive(process):  # pragma: no cover
+                            if self._on_process_down is not None:
+                                self._on_process_down(self, address)
+                            self.process_sub_pool_lost(address)
+                            if self._auto_recover:
+                                await self.recover_sub_pool(address)
+                                if self._on_process_recover is not None:
+                                    self._on_process_recover(self, address)
+                        if recover_events_discovered:
+                            event = self._recover_events.pop(address)
+                            event.set()
+                    except asyncio.CancelledError:
+                        raise
+                    except Exception:
```

Comment:
The original logic is to ignore the exceptions and stop monitoring sub pools, because no one await the task.(the monitor task fails silently). This change is to log the exceptions and continue monitoring sub pools, I think it's better than fails silently.

Reply:
Users need to be informed in time when unexpected errors occurs. More work is needed and a simple error log is far from complete.